packages/components/src/components/Tooltip/TooltipFloatingUi.tsx

```diff
@@ -0,0 +1,190 @@
+    initialOpen?: boolean;
+    placement?: Placement;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+    offset?: number;
+}
+
+export function useTooltip({
+    initialOpen = false,
+    placement = 'top',
+    open: controlledOpen,
+    onOpenChange: setControlledOpen,
+    offset: offsetValue = 10,
+}: TooltipOptions = {}) {
+    const [uncontrolledOpen, setUncontrolledOpen] = React.useState(initialOpen);
+
+    const open = controlledOpen ?? uncontrolledOpen;
+    const setOpen = setControlledOpen ?? setUncontrolledOpen;
+
+    const { delay } = useDelayGroupContext();
+
+    const data = useFloating({
+        placement,
+        open,
+        onOpenChange: setOpen,
+        whileElementsMounted: autoUpdate,
+        middleware: [offset(offsetValue), flip(), shift()],
+    });
+
+    const { context } = data;
+
+    const hover = useHover(context, {
+        move: false,
+        enabled: controlledOpen == null,
+        delay,
+    });
+    const focus = useFocus(context, {
+        enabled: controlledOpen == null,
+    });
+    const dismiss = useDismiss(context);
+    const role = useRole(context, { role: 'tooltip' });
+
+    const interactions = useInteractions([hover, focus, dismiss, role]);
+
+    return React.useMemo(
+        () => ({
+            open,
+            setOpen,
+            ...interactions,
+            ...data,
+        }),
+        [open, setOpen, interactions, data],
+    );
+}
+
+type ContextType = ReturnType<typeof useTooltip> | null;
+
+const TooltipContext = React.createContext<ContextType>(null);
+
+export const useTooltipState = () => {
+    const context = React.useContext(TooltipContext);
+
+    if (context == null) {
+        throw new Error('Tooltip components must be wrapped in <Tooltip />');
+    }
+
+    return context;
+};
+
+export function TooltipFloatingUi({
+    children,
+    ...options
+}: { children: React.ReactNode } & TooltipOptions) {
+    // This can accept any props as options, e.g. `placement`,
+    // or other positioning options.
+    const tooltip = useTooltip(options);
+
+    return <TooltipContext.Provider value={tooltip}>{children}</TooltipContext.Provider>;
+}
+
+export const TooltipTrigger = React.forwardRef<
+    HTMLElement,
+    React.HTMLProps<HTMLElement> & { asChild?: boolean }
+>(function TooltipTrigger({ children, asChild = false, ...props }, propRef) {
+    const state = useTooltipState();
+
+    const childrenRef = (children as any).ref;
+    const ref = useMergeRefs([state.refs.setReference, propRef, childrenRef]);
+
+    // `asChild` allows the user to pass any element as the anchor
+    if (asChild && React.isValidElement(children)) {
+        return React.cloneElement(
+            children,
+            state.getReferenceProps({
+                ref,
+                ...props,
+                ...children.props,
+                'data-state': state.open ? 'open' : 'closed',
+            }),
+        );
+    }
+
+    return (
+        <button
```

Comment:
I really dont like this, I would like to remove it and make `asChild` the only option.

WDYT @jvaclavik - is there any use-case when we dont want to have `asChild` behavior?

Reply:
I removed this.